src/dmclock_server.h

```diff
@@ -112,41 +112,47 @@ namespace crimson {
 
 
     struct RequestTag {
-      double reservation;
-      double proportion;
-      double limit;
-      bool   ready; // true when within limit
-      Time   arrival;
+      double   reservation;
+      double   proportion;
+      double   limit;
+      uint64_t cost;
+      bool     ready; // true when within limit
+      Time     arrival;
 
       RequestTag(const RequestTag& prev_tag,
 		 const ClientInfo& client,
-		 const uint32_t delta,
-		 const uint32_t rho,
+		 const uint64_t delta,
+		 const uint64_t rho,
 		 const Time time,
-		 const double cost = 0.0,
+		 const uint64_t _cost = 1u,
 		 const double anticipation_timeout = 0.0) :
+	cost(_cost),
 	ready(false),
 	arrival(time)
       {
+	assert(cost > 0.0);
```

Comment:
cost is no longer floating point, so `cost > 0` here? (and again on line 180)

for requests that pass cost=0, it's probably better fail with EINVAL error instead of aborting

Reply:
Fixed 0 literal issues.

I don't know that I want to take on improving error returning results in this PR. But I think it's an excellent idea. At the moment there are a lot of asserts that'd need to be tackled.